smartsim/_core/mli/comm/channel/dragon_channel.py

```diff
@@ -133,56 +93,43 @@ def recv(self, timeout: float = 0.001) -> t.List[bytes]:
+                logger.debug(f"DragonCommChannel {self.descriptor} received message")
             except dch.ChannelEmpty:
                 # emptied the queue, ok to swallow this ex
-                logger.debug(f"DragonCommChannel exhausted: {self.descriptor!r}")
-            except dch.ChannelRecvTimeout as ex:
-                logger.debug(f"Timeout exceeded on channel.recv: {self.descriptor!r}")
+                logger.debug(f"DragonCommChannel exhausted: {self.descriptor}")
+            except dch.ChannelRecvTimeout:
+                logger.debug(f"Timeout exceeded on channel.recv: {self.descriptor}")
 
             return messages
 
-    @property
-    def descriptor_string(self) -> str:
-        """Return the channel descriptor for the underlying dragon channel
-        as a string. Automatically performs base64 encoding to ensure the
-        string can be used in a call to `from_descriptor`.
-
-        :returns: String representation of channel descriptor
-        :raises ValueError: If unable to convert descriptor to a string
-        """
-        if isinstance(self._descriptor, str):
-            return self._descriptor
-
-        if isinstance(self._descriptor, bytes):
-            return base64.b64encode(self._descriptor).decode("utf-8")
-
-        raise ValueError(f"Unable to convert channel descriptor: {self._descriptor}")
-
     @classmethod
     def from_descriptor(
         cls,
-        descriptor: t.Union[bytes, str],
+        descriptor: str,
     ) -> "DragonCommChannel":
         """A factory method that creates an instance from a descriptor string.
 
-        :param descriptor: The descriptor that uniquely identifies the resource. Output
-        from `descriptor_string` is correctly encoded.
+        :param descriptor: The descriptor that uniquely identifies the resource.
         :returns: An attached DragonCommChannel
         :raises SmartSimError: If creation of comm channel fails
         """
         try:
-            utf8_descriptor: t.Union[str, bytes] = descriptor
-            if isinstance(descriptor, str):
-                utf8_descriptor = descriptor.encode("utf-8")
-
-            # todo: ensure the bytes argument and condition are removed
-            # after refactoring the RPC models
-
-            actual_descriptor = base64.b64decode(utf8_descriptor)
-            channel = dch.Channel.attach(actual_descriptor)
+            channel = drg_util.descriptor_to_channel(descriptor)
             return DragonCommChannel(channel)
         except Exception as ex:
             raise SmartSimError(
-                f"Failed to create dragon comm channel: {descriptor!r}"
+                f"Failed to create dragon comm channel: {descriptor}"
             ) from ex
+
+    @classmethod
+    def from_local(cls, _descriptor: t.Optional[str] = None) -> "DragonCommChannel":
+        """A factory method that creates a local channel instance.
+
+        :param _descriptor: Unused placeholder
```

Comment:
Is "Unused placeholder" correct?

Reply:
"Optional string that serves as a placeholder for future use"

Reply:
mayhaps since its an optional param? Or maybe add something in the docstring about its intended use in the future

Reply:
it's just a bad decision that i'm stuck with for now. i don't need that string but the protocol is fixed... will come back to it